<?xml version="1.0" encoding="UTF-8"?>
<beans xmlns="http://www.springframework.org/schema/beans"
       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
       xmlns:mvc="http://www.springframework.org/schema/mvc"
       xsi:schemaLocation="http://www.springframework.org/schema/beans
        http://www.springframework.org/schema/beans/spring-beans.xsd
        http://www.springframework.org/schema/mvc
        http://www.springframework.org/schema/mvc/spring-mvc.xsd">

    <mvc:annotation-driven/>

    <mvc:resources mapping="/resources/**" location="/resources/"/>

    <!--<context:component-scan base-package="ru.innopolis.uni.course3.controller"/>-->
    <bean id="BookController" class="ru.innopolis.uni.course3.controller.BookController">
        <constructor-arg ref="BookService"/>
    </bean>
    <bean id="UserController" class="ru.innopolis.uni.course3.controller.UserController">
        <constructor-arg index="0" ref="UserService"/>
        <constructor-arg index="1" ref="PasswordAuthentication"/>
        <constructor-arg index="2" ref="myUserDetailsService"/>
    </bean>
    <!--<bean id="ExceptionHandlingController" class="ru.innopolis.uni.course3.controller.deprecated.ExceptionHandlingController" />-->

    <bean id="viewResolver" class="org.springframework.web.servlet.view.InternalResourceViewResolver" >
        <property name="viewClass" value="org.springframework.web.servlet.view.JstlView" />
        <property name="prefix">
            <value>/WEB-INF/views/</value>
        </property>
        <property name="suffix">
            <value>.jsp</value>
        </property>
    </bean>

    <bean id="simpleMappingExceptionResolver" class="org.springframework.web.servlet.handler.SimpleMappingExceptionResolver">
        <property name="exceptionMappings">
            <map>
                <entry key="ru.innopolis.uni.course3.exception.WrongProcessingOfBookException" value="wrong" />
                <entry key="ru.innopolis.uni.course3.exception.WrongProcessingOfUserException" value="wrong"/>
                <entry key="org.hibernate.exception.ConstraintViolationException" value="wrong"/>
                <entry key="org.springframework.dao.EmptyResultDataAccessException" value="wrong"/>
                <entry key="javax.persistence.OptimisticLockException" value="wrong"/>
            </map>
        </property>
        <property name="defaultErrorView" value="wrong"/>
        <property name="exceptionAttribute" value="exception"/>
    </bean>

</beans>